realm/kotlin-extensions/src/androidTestObjectServer/kotlin/io/realm/kotlin/KotlinSyncedRealmTests.kt

```diff
@@ -0,0 +1,55 @@
+package io.realm.kotlin
+
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import io.realm.Realm
+import io.realm.SyncConfiguration
+import io.realm.SyncManager
+import io.realm.TestSyncConfigurationFactory
+import io.realm.objectserver.utils.Constants
+import io.realm.util.SyncTestUtils
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class KotlinSyncedRealmTests {
+
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+
+    private lateinit var realm: Realm
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getTargetContext())
+        val user = SyncTestUtils.createTestUser()
+        realm = Realm.getInstance(configFactory.createSyncConfigurationBuilder(user, Constants.DEFAULT_REALM).build())
+    }
+
+    @After
+    fun tearDown() {
+        realm.close()
+    }
+
+    @Test
+    fun synsSession() {
+        assertEquals(SyncManager.getSession(realm.configuration as SyncConfiguration), realm.syncSession)
+    }
+
+    @Test
+    fun syncSession_throwsForNonSyncRealm() {
```

Comment:
Maybe add a test that throws if the SyncConf doesn't correspond to an existing SyncSession